frontend/packages/knative-plugin/src/utils/fetch-knative-resources.ts

```diff
@@ -0,0 +1,57 @@
+import * as _ from 'lodash';
+import { K8sResourceKind } from '@console/internal/module/k8s';
+
+const IS_KNATIVE = 'serving.knative.dev/service';
+
+type KnativeItem = {
+  ksroutes: K8sResourceKind[];
+  configurations: K8sResourceKind[];
+  revisions: K8sResourceKind[];
+};
+
+const isKnativeDeployment = (dc: K8sResourceKind): boolean => {
+  return !!(dc.metadata && dc.metadata.labels && dc.metadata.labels[IS_KNATIVE]);
+};
+
+const getKSRoute = (dc: K8sResourceKind, { ksroutes }): K8sResourceKind[] => {
+  let routeResource = [];
+  if (isKnativeDeployment(dc)) {
+    routeResource = _.filter(ksroutes.data, (routeConfig: K8sResourceKind) => {
+      return dc.metadata.labels[IS_KNATIVE] === _.get(routeConfig, 'metadata.name');
+    });
+  }
+  return routeResource;
+};
+
+const getConfigurations = (dc: K8sResourceKind, { configurations }): K8sResourceKind[] => {
+  let configurationResource = [];
+  if (isKnativeDeployment(dc)) {
+    configurationResource = _.filter(configurations.data, (config: K8sResourceKind) => {
+      return dc.metadata.labels[IS_KNATIVE] === _.get(config, 'metadata.name');
+    });
+  }
+  return configurationResource;
+};
+
+const getRevisions = (dc: K8sResourceKind, { revisions }): K8sResourceKind[] => {
+  let revisionResource = [];
+  if (isKnativeDeployment(dc)) {
+    revisionResource = _.filter(revisions.data, (revision: K8sResourceKind) => {
+      return dc.metadata.ownerReferences[0].uid === revision.metadata.uid;
+    });
+  }
+  return revisionResource;
+};
+
+const fecthKnativeServingResources = (dc: K8sResourceKind, props): KnativeItem => {
+  const ksroutes = getKSRoute(dc, props);
+  const configurations = getConfigurations(dc, props);
+  const revisions = getRevisions(dc, props);
+  return {
+    ksroutes,
+    configurations,
+    revisions,
+  };
+};
+
+export default fecthKnativeServingResources;
```

Comment:
For utilities, let's stick to named exports.
Also, fetch is associated with network calls but there is no such call here. Prefer a different name.